lib/ansible/modules/windows/win_get_url.ps1

```diff
@@ -19,120 +20,186 @@
 # WANT_JSON
 # POWERSHELL_COMMON
 
+
+Function CheckModified-File($url, $dest, $headers, $credentials, $timeout, $use_proxy, $proxy) {
```

Comment:
Inside `CheckModified-File` and `Download-File` you have an if statement `if (-not $use_proxy) { $webRequest.Proxy = $null } else { $webRequest.Proxy = $proxy }`. Outside of this you have already set the value of `$proxy` to `$null`. Why don't you just pass that null value and always set `$webRequest.Proxy = $proxy`. Saves another if statement inside each function and removes another variable you are passing to them.

I don't see any docs that state it will use the current system's proxy if not configured, I was under the assumption that if you you specify anything under the `Proxy` property then no proxy will be used. This is why I suggested that code block where it will use the default system proxy if `$use_proxy` is true but no URL is defined.

Note I don't use proxies so cannot verify this but can't find any evidence to back up the opposite.